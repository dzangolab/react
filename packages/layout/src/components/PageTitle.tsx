import React from "react";

interface Properties {
  title?: string;
  toolbar?: React.ReactNode;
}

const PageTitle = ({ title, toolbar }: Properties) => {
  return (
    <div className="page-title">
      {title && <h1>{title}</h1>}

      {toolbar && <div className="toolbar">{toolbar}</div>}
    </div>
  );
};

export default PageTitle;
