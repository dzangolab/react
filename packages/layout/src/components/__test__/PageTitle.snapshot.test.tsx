import renderer from "react-test-renderer";
import { expect, test } from "vitest";

import PageTitle from "../PageTitle";

function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON();

  expect(result).toBeDefined();
  expect(result).not.toBeInstanceOf(Array);

  return result as renderer.ReactTestRendererJSON;
}

test("Component matches snapshot", () => {
  const component = renderer.create(
    <PageTitle title="Test title" toolbar={<div>Toolbar content</div>} />
  );
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
