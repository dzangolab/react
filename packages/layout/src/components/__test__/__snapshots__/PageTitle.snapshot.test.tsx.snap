// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component matches snapshot 1`] = `
<div
  className="page-title"
>
  <h1>
    Test title
  </h1>
  <div
    className="toolbar"
  >
    <div>
      Toolbar content
    </div>
  </div>
</div>
`;
